cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/task/ClusterPollSubtask.java

```diff
@@ -38,6 +45,19 @@ public ClusterPollSubtask(Provisioner provisioner, ProvisionerContext provisione
   @Override
   protected Cluster execute(Cluster cluster) throws Exception {
     ClusterStatus currentStatus = provisioner.getClusterStatus(provisionerContext, cluster);
+
+    if (cluster.getStatus() == ClusterStatus.FAILED) {
```

Comment:
This class is used to poll for state until it changes, so it's not really a good place to put this log. It's extra work that happens in every poll, and it's also possible somebody uses this class to poll when the cluster moves from failed to some other state, in which case this would log repeatedly.

Instead, this should be logged only when a cluster transitions into the failed state and moves to delete the cluster. Looks like this can be done in ProvisionTask.createPollingCreateSubtask().

Reply:
Good point. Moved it there.